src/modules/hunting/etcd.py

```diff
@@ -81,66 +50,57 @@ def db_keys_write_access(self):
     """
-    # TODO:
-    # Check the etcd hunter on a remote cluster! (currently everything was checked only at 127.0.0.1:2379)
+
     def __init__(self, event):
         self.event = event
         self.version_evidence = ''
         self.keys_evidence = ''
+        self.protocol = 'https'
 
     def db_keys_disclosure(self):
-        logging.debug(self.event.host)
-        logging.debug("Passive hunter is attempting to read etcd keys remotely")
-        r_secure = "https://{host}:{port}/v2/keys".format(host=self.event.host, port=2379)
-        r_not_secure = "http://{host}:{port}/v2/keys".format(host=self.event.host, port=2379)
-        res = helperFuncDo2Requests(r_secure, r_not_secure)
-        if res:
-            self.keys_evidence = res.content
-            return True
-        return False
+        logging.debug(self.event.host + " Passive hunter is attempting to read etcd keys remotely")
+        r = requests.get(
+            "{protocol}://{host}:{port}/v2/keys".format(protocol=self.protocol, host=self.event.host, port=2379),
+            verify=False)
+        self.keys_evidence = r.content if r.status_code == 200 and r.content != '' else False
+        return self.version_evidence
 
     def version_disclosure(self):
-        logging.debug(self.event.host)
-        logging.debug("Passive hunter is attempting to check etcd version remotely")
-        r_secure = "https://{host}:{port}/version".format(host=self.event.host, port=2379)
-        r_not_secure = "http://{host}:{port}/version".format(host=self.event.host, port=2379)
-        res = helperFuncDo2Requests(r_secure, r_not_secure)
-        if res:
-            self.version_evidence = res.content
-            return True
-        return False
+        logging.debug(self.event.host + " Passive hunter is attempting to check etcd version remotely")
+        r = requests.get(
+            "{protocol}://{host}:{port}/version".format(protocol=self.protocol, host=self.event.host, port=2379),
+            verify=False)
+        self.version_evidence = r.content if r.status_code == 200 and r.content != '' else False
+        return self.version_evidence
 
     def unauthorized_access(self):
-        logging.debug(self.event.host)
-        logging.debug("Passive hunter is attempting to access etcd without authorization")
-        r_not_secure = "http://{host}:{port}/version".format(host=self.event.host, port=2379)
-        res = helperFuncDo2Requests(r_not_secure, r_not_secure)  # We don't have to do 2 requests this time
-        if res:
-            return True
-        return False
+        logging.debug(self.event.host + " Passive hunter is attempting to access etcd without authorization")
+        r = requests.get("http://{host}:{port}/version".format(host=self.event.host, port=2379), verify=False)
+        return r.content if r.status_code == 200 and r.content != '' else False
 
     def execute(self):
+        if self.unauthorized_access():  # decide between http and https protocol
```

Comment:
This approach to trying insecure first and then falling back to https is really neat, I like it!

Reply:
Thanks you!